File: gcd/implementation/cpp/gcd.cpp
#include <iostream>

using namespace std;

int64_t gcd(int64_t a, int64_t b);

int main(){
    int64_t a, b;
    cout<<"Enter the numbers for which you want to calculate GCD :"<<endl;
    cin>>a>>b;

    cout<<"GCD of "<<a<<" and "<<b<<" is : "<<gcd(a, b);

    return 0;
}

int64_t gcd(int64_t a, int64_t b){
    if(a == 0 || a == b){
        return b;
    }else if(b == 0){
        return a;
    }
    else{
        if(a > b){
            return gcd(b, a % b);
        }else{
            return gcd(a, b % a);
        }
    }
}

File: gcd/implementation/java/GCD.java
import java.util.Scanner;
import java.math.BigInteger;

public class GCD {
    static BigInteger zero = new BigInteger("0");
    public static void main(String args[]){
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the numbers for which you want to calculate GCD :");

        BigInteger a = new BigInteger(input.next());
        BigInteger b = new BigInteger(input.next());

        System.out.println("GCD of " + a + " and " + b + " is : " + gcd(a, b));
        input.close();
    }

    static BigInteger gcd(BigInteger a, BigInteger b) {
        if(a.equals(zero) || a.equals(b)){
            return b;
        }else if(b.equals(zero)){
            return a;
        }else{
            if(a.compareTo(b) == 1){
                return gcd(b, a.divideAndRemainder(b)[1]);
            }else{
                return gcd(a, b.divideAndRemainder(a)[1]);
            }
        }
    }
}
